### PR TITLE
SAM debug: target=code: require TypeScript on host system

### PR DESCRIPTION
**Pull request by gsweetwood**
## Problem
If the TypeScript compiler is not installed and a local invoke is called where the target is the source code, the app will not fail until the invoke step and the error is not clear.
## Solution
Check for TypeScript compiler before attempting to compile and show a clear error if it is not present.

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/shared/sam/debugger/typescriptSamDebug.ts [hunk 1]
```diff
@@ -118,6 +118,10 @@ export async function makeTypescriptConfig(config: SamLaunchRequestArgs): Promis
  */
 async function compileTypeScript(config: NodejsDebugConfiguration): Promise<void> {
     if (config.invokeTarget.target === 'code') {
+        const tscResponse = await new ChildProcess(true, 'tsc -v').run()
```
Comment (justinmk3) [inline src/shared/sam/debugger/typescriptSamDebug.ts:121]: always pass args separately / as a list.

```suggestion
        const tscResponse = await new ChildProcess(true, 'tsc', undefined, '-v').run()
```

File: src/shared/sam/debugger/typescriptSamDebug.ts [hunk 2]
```diff
@@ -118,6 +118,10 @@ export async function makeTypescriptConfig(config: SamLaunchRequestArgs): Promis
...
+        if (!tscResponse.stdout.startsWith('Version') ) {
```
Comment (justinmk3) [inline src/shared/sam/debugger/typescriptSamDebug.ts:122]: should also check exitCode() === 0

File: src/shared/sam/debugger/typescriptSamDebug.ts [hunk 3]
```diff
@@ -118,6 +118,10 @@ export async function makeTypescriptConfig(config: SamLaunchRequestArgs): Promis
...
+            throw new Error('The TypeScript compiler was not found. To install globally on your machine, run "npm install -g typescript" in the terminal.')
```
Comment (justinmk3) [inline src/shared/sam/debugger/typescriptSamDebug.ts:123]: If we are going to give instructions then we are in localization territory. If we are going to surface this as Error() let's keep it precise yet concise.
```suggestion
            throw new Error('TypeScript compiler "tsc" not found.')
```

File: src/shared/sam/debugger/typescriptSamDebug.ts [hunk 4]
```diff
@@ -118,6 +118,10 @@ export async function makeTypescriptConfig(config: SamLaunchRequestArgs): Promis
  */
 async function compileTypeScript(config: NodejsDebugConfiguration): Promise<void> {
     if (config.invokeTarget.target === 'code') {
+        const tscResponse = await new ChildProcess(true, 'tsc', undefined, '-v').run()
+        if (tscResponse.exitCode === 0 || !tscResponse.stdout.startsWith('Version')) {
```
Comment (justinmk3) [inline src/shared/sam/debugger/typescriptSamDebug.ts:122]: exitCode 0 means success, so this should check `!==`

Comment (gsweetwood) [reply src/shared/sam/debugger/typescriptSamDebug.ts]: good catch, that was a typo

**Pull request merged**